pkg/kubefedctl/federate/federate.go

```diff
@@ -192,6 +222,42 @@ func (j *federateResource) Run(cmdOut io.Writer, config util.FedConfig) error {
 	return CreateResources(cmdOut, hostConfig, artifactsList, j.FederationNamespace, j.enableType, j.DryRun)
 }
 
+func federateResources(resources []*unstructured.Unstructured) ([]*unstructured.Unstructured, error) {
+	var federatedResources []*unstructured.Unstructured
+	for _, targetResource := range resources {
+		// A Group, a Version and a Kind is sufficient for API Resource definition.
+		gvk := targetResource.GroupVersionKind()
+
+		// Construct an API Resource from above info.
+		// TODO(irfanurrehman) Should we depend on the lookup from the
```

Comment:
I think the answer to this question is yes, for the same reasons that we've previously discussed regarding `kubefedctl federate ... -o yaml`.  If a type config is not available from the api, by all means use a generated one.  But if one is available, ignoring it risks generating yaml that will not be compatible with the FTC in the API.

Reply:
To point 1, I don't think we have much visibility on whether conversion will be useful for federated types.  I'm not confident that internal API conversion is going to have the same utility for other types as for kubefed, since a resource can only be propagated to member clusters at a single version and I have yet to hear a reason why that version shouldn't be the latest available.

Point 2 suggests that federate should compare the version of a resource targeted for conversion with the version of the type configured in a deployed FTC.  federate should error out if the version of the resource targeted for conversion is greater than that of the type configured for the FTC.  If this occurs, a helpful message could be provided that indicates how to upgrade the type targeted by the FTC.

Point 3 suggests to me that the user should be able to control whether federate uses an apiserver or not to convert file intput to yaml.  If their intent is to create federated content with compatibility for control planes whose types are enabled with defaults, then they don't need to use an apiserver.  If their intent is to create federated content with compatibility for a control plane whose types have non-default naming or groups, then they would want to use the apiserver hosting that control plane.  Maybe `--use-apiserver`?  

Speaking of undocumented complexity, are you working on updating the userguide to reflect the work you've been doing on `federate` and to document issues like these?

Reply:
I am working on documentation of federate feature. I would prefer to work on the details we discussed here outside this PR. Created #864 for that.

Reply:
Fine to file it separately so long as you document the caveats of what is implemented now.  I would prefer that you don't pursue #864 until post-beta in any case.